services/user-svc/internal/organization/organization.go

```diff
@@ -794,3 +788,84 @@ func AddUserToOrganization(ctx context.Context, userId uuid.UUID, organizationId
 
 	return nil
 }
+
+func (s ServiceServer) CreatePersonalOrganization(ctx context.Context, _ *pb.CreatePersonalOrganizationRequest) (*pb.CreatePersonalOrganizationResponse, error) {
+	kc, err := hwkc.NewClientFromEnv(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	userID, err := common.GetUserID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	userClaims, err := common.GetAuthClaims(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	organisations, err := kc.GetOrganizationsForUserById(userID)
+	if err != nil {
+		return nil, err
+	}
+
+	personalOrganizations := hwutil.Filter(organisations, func(organization *hwkc.Organization) bool {
+		return organization.Attributes.IsPersonal
+	})
+
+	if len(personalOrganizations) > 0 {
+		organizationID, err := uuid.Parse(*personalOrganizations[0].ID)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse id from keycloak for organization: %w", err)
+		}
+
+		return &pb.CreatePersonalOrganizationResponse{
+			Id: organizationID.String(),
+		}, nil
+	}
+
+	organizationName := fmt.Sprintf("Personal organization of %s", userClaims.Name)
+
+	userRepo := user_repo.New(hwdb.GetDB())
+
+	userResult, err := hwdb.Optional(userRepo.GetUserById)(ctx, userID)
+	err = hwdb.Error(ctx, err)
+	if err != nil {
+		return nil, err
+	} else if userResult == nil {
+		hash := sha256.Sum256([]byte(userID.String()))
+		avatarUrl := fmt.Sprintf("%s%s", "https://source.boringavatars.com/marble/128/", hex.EncodeToString(hash[:]))
+
+		_, err = userRepo.CreateUser(ctx, user_repo.CreateUserParams{
+			ID:        userID,
+			Email:     userClaims.Email,
+			Nickname:  userClaims.PreferredUsername,
+			Name:      userClaims.Name,
+			AvatarUrl: &avatarUrl,
+		})
+		err = hwdb.Error(ctx, err)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
nit (also at some other places)
```suggestion
		if err := hwdb.Error(ctx, err); err != nil {
			return nil, err
		}
```